### ISSUE TITLE
Private IP being assigned to db_system is null

### ISSUE THREAD

**Issue by kaustubh77** [terraform-providers/terraform-provider-oci#1307]
Following is the code that I used for creating a db_system : 
```
resource "oci_database_db_system" "db_system_2" {
  availability_domain = data.oci_identity_availability_domain.ad.name
  compartment_id = "${var.compartment_id}"
  database_edition = "${var.database_edition}"
  db_home {
    database {
      admin_password = "${var.admin_password}"
      character_set = "${var.character_set}"
      db_name = "test"
      db_workload = "${var.db_workload}"
      freeform_tags = { "Project" = "${var.project_name}"}
      ncharacter_set = "${var.ncharacter_set}"
      pdb_name = "${var.pdb_name}"
    }
    db_version = "21.1.0.0"
  }
  cpu_core_count = "4"
  hostname = "test2"
  shape = "${var.shape}"
  ssh_public_keys = ["${file(var.ssh_public_keys)}"]
  subnet_id = "${var.subnet_id}"
  node_count = "${var.node_count}"
  data_storage_size_in_gb = "${var.data_storage_size_in_gb}"
  display_name = "test"

  connection {
    type        = "ssh"
    host        = "${self.private_ip}"
    user        = "opc"
    private_key = "${file(var.ssh_private_key_file)}"
  }

  provisioner "file" {
    source      = "~/Downloads/test/bash.sh"
    destination = "/home/opc/bash.sh"
  }

  provisioner "remote-exec" {
    inline = [
      "chmod 777 /home/opc/bash.sh",
      "/home/opc/bash.sh",
    ]
  }
}
```
However, it throws me this error - 

> Error: host for provisioner cannot be empty

The db_system is created successfully, I can see it on the console website. The error comes from the provisioner step. According to terraform state file the private IP is being set to null. And that is the reason it gives the error "host cannot be empty". However, I can see the private IP on the console website. What am I missing here? The terraform documentation of db_system mentions that 
> If you don't specify a value, Oracle automatically assigns a private IP address from the subnet.

Any kind of help is appreciated. Thank you

**Comment by zexinwanoci** [comment]
Hello, I don't think   connection {
    type        = "ssh"
    host        = "${self.private_ip}"
    user        = "opc"
    private_key = "${file(var.ssh_private_key_file)}"
  }

  provisioner "file" {
    source      = "~/Downloads/test/bash.sh"
    destination = "/home/opc/bash.sh"
  }

  provisioner "remote-exec" {
    inline = [
      "chmod 777 /home/opc/bash.sh",
      "/home/opc/bash.sh",
    ]
  }

**Comment by zexinwanoci** [comment]
Hello, I have not seem   connection {
    type        = "ssh"
    host        = "${self.private_ip}"
    user        = "opc"
    private_key = "${file(var.ssh_private_key_file)}"
  }

  provisioner "file" {
    source      = "~/Downloads/test/bash.sh"
    destination = "/home/opc/bash.sh"
  }

  provisioner "remote-exec" {
    inline = [
      "chmod 777 /home/opc/bash.sh",
      "/home/opc/bash.sh",
    ]
  }
being used inside resource "oci_database_db_system". Can you check our examples and fix that config file?

**Comment by kaustubh77** [comment]
Thanks @zexinwanoci for answering. I used the above mentioned way of writing the connection block from some github repo. Anyway, I got the solution to my problem. The problem was happening because the private_ip being assigned to my db_system is set to  null as per the terraform.state file when the db system got provisioned. The db nodes are not immediately provisioned though, so the private IP was being set to null, this is what I think, correct me if I am wrong. What worked for me was getting db nodes associated with the db system and from the db nodes getting the vnics and using vnic we can access the private IPs. This 3 step process worked for me. Thanks for responding so promptly!
